bgpd/bgp_nb_config.c

```diff
@@ -123,7 +123,12 @@ int bgp_router_create(struct nb_cb_create_args *args)
 		if (is_new_bgp && inst_type == BGP_INSTANCE_TYPE_DEFAULT)
 			vpn_leak_postchange_all();
 
-		if (inst_type == BGP_INSTANCE_TYPE_VRF)
+		/*
+		 * Check if we need to export to other VRF(s).
+		 * Leak the routes to importing bgp vrf instances,
+		 * only when new bgp vrf instance is configured.
+		 */
+		if (ret != BGP_INSTANCE_EXISTS)
```

Comment:
Does it mean, that routes will be exported if the instance exists and is route-server (VIEW)?

Reply:
These changes are removed from this commit. Since it requires topotest changes. It will be committed later.

Reply:
This check to avoid releaking (export), every time we configure "router bgp xxx".
We want to releak only when it is new instance; else for existing bgp instance we do not.

Reply:
So, to be sure, we don't care here if we will export even if this is a non-VRF instance (view)? Because previously we checked ONLY BGP_INSTANCE_TYPE_VRF. Is that fine?